shared/src/metabase/shared/models/visualization_settings.cljc

```diff
@@ -382,7 +383,8 @@
    :prefix            ::prefix
    :suffix            ::suffix
    :view_as           ::view-as
-   :link_text         ::link-text})
+   :link_text         ::link-text
+   :show_mini_bar     ::show-mini-barchart})
```

Comment:
In #16972 I looked through the viz settings on the frontend and added a bunch of additional settings that seemed to be missing here. Would it be worth adding those in this PR instead?

https://github.com/metabase/metabase/blob/export-formatting/shared/src/metabase/shared/models/visualization_settings.cljc#L373-L392

Related: should we try to automatically convert the DB form into the normalized form as a fallback in case a key isn't in this map? Otherwise it seems like the key just becomes `nil` after the conversion.

Reply:
> This LGTM. Is there anywhere we can add some comments or docstrings explaining what's going on/why (e.g. in `resolve-table-columns`?) Just so someone else looking thru the code in the future knows why we're doing what we're doing

Yes, I will go through and add docstrings to the `load.clj` stuff before submitting.

Reply:
Yeah I realized that this was going to conflict with the stuff you're working on 😢 

I wanted to keep this one fairly limited and not cherry pick too much in, especially considering you're adding independent tests for all of this.  What do you think?

Regarding the fallback thing, let me restate this in the way I understand it.  If we don't find one of the keys during `db->norm` conversion (in any kind of map), we should just preserve it as-is (i.e. the unqualified keyword).  Is that what you mean?  If so, I would tend to agree, although I would prefer to keep that behavior to your WIP patch rather than bring it in here (see above).

Reply:
Sure, I can keep those in my PR.

Re: fallback, could we just convert the unqualified keys to strings, change `_` to `-`, and then convert back into qualified keywords? That would work for all the current keys that we have mappings for.

Reply:
Yes, I think that's a good approach.  But let's tackle that separately from this patch (either in a standalone PR, or as part of what you're doing).